src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -2272,7 +2272,8 @@ public List<Setting<?>> getSettings() {
                     ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED,
                     ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED_DEFAULT,
                     Property.NodeScope,
-                    Property.Filtered
+                    Property.Filtered,
+                    Property.Dynamic
```

Comment:
Has this been tested? IMHO, it is not sufficient to add `Property.Dynamic` to a `Setting` object to make it dynamic; you also have to listed to settings updates. The way it is implemented currently, `PrivilegesEvaluator` will still read from the immutable `Settings` object that will never change.

See here for a minimal implementation:

https://github.com/opensearch-project/OpenSearch/blob/5e6ee972a7f0eb7b457bec50ed49f70ed52a7f72/server/src/main/java/org/opensearch/action/support/AutoCreateIndex.java#L76-L79

Reply:
@nibix @kumjiten Doing manual testing, I found that I could not view or update this setting with the cluster settings API. After discussing with @cwperks, we found that:

- I needed to remove `Property.FILTERED` for my setting: f8169f81312dab359440923ef7d129646cdd90b6
- There are two instances of `getSettingsFilter` which seem to exclude all security plugin settings (`plugins.security.*`) from management by the cluster settings API by default. We updated these methods to change the default filters to something less restrictive: c0d8e2ad271d87d8ad3e33f665867ce2dfce2de2, since properties can also be filtered by using `Property.FILTERED`